core-java-modules/core-java-regex/src/test/java/com/baeldung/regex/findnumbers/FindNumbersUnitTest.java

```diff
@@ -0,0 +1,128 @@
+package com.baeldung.regex.findnumbers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.DoubleStream;
+import java.util.stream.LongStream;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit Test to find Integers, Decimal Numbers, and Scientific Notation and Hexadecimal Numbers in a String
+ */
+class FindNumbersUnitTest {
+
+    private static List<String> findIntegers(String stringToSearch) {
+        Pattern integerPattern = Pattern.compile("-?\\d+");
+        Matcher matcher = integerPattern.matcher(stringToSearch);
+
+        List<String> integerList = new ArrayList<>();
+        while (matcher.find()) {
+            integerList.add(matcher.group());
+        }
+
+        return integerList;
+    }
+
+    private static List<String> findDecimalNums(String stringToSearch) {
+        Pattern decimalNumPattern = Pattern.compile("-?\\d+(\\.\\d+)?");
+        Matcher matcher = decimalNumPattern.matcher(stringToSearch);
+
+        List<String> decimalNumList = new ArrayList<>();
+        while (matcher.find()) {
+            decimalNumList.add(matcher.group());
+        }
+
+        return decimalNumList;
+    }
+
+    @Test
+    void givenStrOfAllDigits_whenRegexMatchByInt_thenWholeStrMatchedAsOneInt() {     
+        List<String> integersFound = findIntegers("970987678607608");
+
+        assertThat(integersFound).contains("970987678607608");
+    }
+
+    @Test
+    void givenStrWithIntegersSepByPeriods_whenRegexMatchByInt_thenExpectedIntsFound() {
+        List<String> integersFound = findIntegers("3453..5.-23532...32432.-2363.3454......345.-34.");
+
+        assertThat(integersFound).contains("3453", "5", "-23532", "32432", "-2363", "3454", "345", "-34");
+    }
+
+    @Test
+    void givenStrWithIntegersSepByNonDigits_whenRegexMatchByInt_thenExpectedIntsFound() {
+        List<String> integersFound = findIntegers("646lkl~4-53l-k34.fdsf.-ds-35.45f9wg3868*()(k;-95786fsd79-86");
+
+        assertThat(integersFound).contains("646", "4", "-53", "34", "-35", "45", "9", "3868", "-95786", "79", "-86");
+    }
+    
+    @Test
+    void givenStrOfAllDigits_whenRegexMatchByDecNum_thenWholeStrMatchedAsOneDecimalNumber() {
+        List<String> decimalNumsFound = findDecimalNums("970987678607608");
+        
+        assertThat(decimalNumsFound).contains("970987678607608");
+    }
+    
+    @Test
+    void givenStrOfDecNumsSepByNonDigits_whenRegexMatchByDecNum_thenExpectedNumsFound() {
+        List<String> decimalNumsFound = findDecimalNums(".7854.455wo.rdy(do.g)-3.-553.00.53;good^night%o3456sdcardR%3567.4%£cat");
+        
+        assertThat(decimalNumsFound).contains("7854.455", "-3", "-553.00", "53", "3456", "3567.4");
```

Comment:
minor - these should probably be `containsExactly` to ensure there's no extra stuff

Reply:
across all the `contains` calls